### PR TITLE
[change] switch from xml to java configuration

### LINKED ISSUE #72: Switch from xml to java configuration
The xml configuration should be removed and migrated to java configuration. The Spring Framework provides this since version 3.1 and this can be done as a first step to [Spring Boot](http://projects.spring.io/spring-boot/). For reference, these steps can be used to apply also to the [OSIAM server](https://github.com/osiam/server).

### PR DESCRIPTION
**Pull request by dacrome**
Closes #72

### CODE REVIEW

File: src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java [hunk 1]
```diff
@@ -122,12 +117,11 @@ public void index(HttpServletResponse response) throws IOException {
...
      *            The new email address value
      * @return The HTTP status code
-     * @throws IOException
-     * @throws MessagingException
+     * @throws IOException If the updated user object can't mapped to json
      */
     @RequestMapping(method = RequestMethod.POST, value = "/change", produces = "application/json")
     public ResponseEntity<String> change(@RequestHeader("Authorization") final String authorization,
-            @RequestParam("newEmailValue") final String newEmailValue) throws IOException, MessagingException {
+                                         @RequestParam("newEmailValue") final String newEmailValue) throws IOException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java [hunk 2]
```diff
@@ -201,11 +195,13 @@ private User getUpdatedUserForEmailChange(String token, String newEmail, OneTime
...
      *            The previously generated confirmation token from the confirmation email
      * @return The HTTP status code and the updated user if successful
+     * @throws IOException
+     *
      */
     @RequestMapping(method = RequestMethod.POST, value = "/confirm", produces = "application/json")
     public ResponseEntity<String> confirm(@RequestHeader("Authorization") final String authorization,
-            @RequestParam("userId") final String userId,
-            @RequestParam("confirmToken") final String confirmationToken) throws IOException, MessagingException {
+                                          @RequestParam("userId") final String userId,
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java [hunk 3]
```diff
@@ -201,11 +195,13 @@ private User getUpdatedUserForEmailChange(String token, String newEmail, OneTime
...
+                                          @RequestParam("confirmToken") final String confirmationToken) throws IOException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java [hunk 4]
```diff
@@ -271,7 +267,7 @@ private User getUpdatedUserForEmailChange(String token, String newEmail, OneTime
 
         try {
             renderAndSendEmailService.renderAndSendEmail("changeemailinfo", config.getFromAddress(), oldEmail.get()
-                    .getValue(),
+                            .getValue(),
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/ChangeEmailController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 5]
```diff
@@ -104,7 +100,7 @@
      */
     @RequestMapping(value = "/lost/{userId}", method = RequestMethod.POST, produces = "application/json")
     public ResponseEntity<String> lost(@RequestHeader("Authorization") final String authorization,
-            @PathVariable final String userId) throws IOException, MessagingException {
+                                       @PathVariable final String userId) throws IOException, MessagingException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 6]
```diff
@@ -145,7 +141,7 @@
 
         try {
             renderAndSendEmailService.renderAndSendEmail("lostpassword", config.getFromAddress(), email.get()
-                    .getValue(), locale,
+                            .getValue(), locale,
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 7]
```diff
@@ -167,11 +163,11 @@
      */
     @RequestMapping(value = "/lostForm", method = RequestMethod.GET)
     public void lostForm(@RequestParam("oneTimePassword") String oneTimePassword,
-            @RequestParam("userId") String userId,
-            HttpServletResponse response) throws IOException {
+                         @RequestParam("userId") String userId,
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 8]
```diff
@@ -167,11 +163,11 @@
...
+                         HttpServletResponse response) throws IOException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 9]
```diff
@@ -206,8 +202,8 @@ public void lostForm(@RequestParam("oneTimePassword") String oneTimePassword,
      */
     @RequestMapping(value = "/change", method = RequestMethod.POST, produces = "application/json")
     public ResponseEntity<String> changePasswordByUser(@RequestHeader("Authorization") final String authorization,
-            @RequestParam String oneTimePassword,
-            @RequestParam String newPassword) throws IOException {
+                                                       @RequestParam String oneTimePassword,
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

File: src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java [hunk 10]
```diff
@@ -206,8 +202,8 @@ public void lostForm(@RequestParam("oneTimePassword") String oneTimePassword,
...
+                                                       @RequestParam String newPassword) throws IOException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/LostPasswordController.java]: wrong indentation

**Comment by dacrome** [comment]
updated.

**Pull request merged**